src/setup.ts

```diff
@@ -0,0 +1,16 @@
+import { ApolloServer } from "apollo-server";
+import { typeDefs } from "./schema";
+import { resolvers } from "./resolvers";
+import { ConfigAppDataSource } from "./data-source";
+
+export async function setup() {
+  await ConfigAppDataSource();
+
+  const server = new ApolloServer({ typeDefs, resolvers });
+
+  server
+    .listen(parseInt(process.env.ENV_SERVER_PORT!))
+    .then(async ({ url }) => {
+      console.log(`🚀  Server ready at ${url}`);
+    });
```

Comment:
Aqui pode ter um pequeno problema: a função `setup` vai finalizar sua execução assim que o comando `server.listen.then` for finalizado, ou seja, antes de o listen resolver. Dessa forma, pode ser que o que seja chamado depois do setup pode acabar fazendo algo sem que o listen tenha resolvido ainda. Outro ponto é que se o listen for rejeitado, ele vai jogar um UnhadledPromiseError em vez do erro original do listen.

O ideal seria ou você retornar a promise listen ou aguardar sua execução com await:

```ts
return server
    .listen(parseInt(process.env.ENV_SERVER_PORT!))
    .then(async ({ url }) => {
      console.log(`🚀  Server ready at ${url}`);
    });
```

```ts
const { url } = await server.listen(parseInt(process.env.ENV_SERVER_PORT!))
console.log(`🚀 Server ready at ${url}`);
```